### PR TITLE
[orbis-kenel] implement sys_kenv

### PR DESCRIPTION
**Pull request by viniciuslrangel**
I'm still learning the code base. If you can help me if I got anything wrong or suggestions, I want to learn and to contribute even more

### CODE REVIEW

**Review by DHrpcs3 — CHANGES_REQUESTED** [review]
Mostly looks good. Thank you

**Comment by viniciuslrangel** [comment]
Thank you for the feedback

File: orbis-kernel/src/sys/sys_environment.cpp [hunk 1]
```diff
@@ -1,7 +1,85 @@
...
+        len -= entry;
+        done += entry;
+      }
+      needed += entry;
+    }
+
+    if (done == needed) {
+      return {};
+    }
+    return {static_cast<ErrorCode>(-needed)};
```
Comment (DHrpcs3) [inline orbis-kernel/src/sys/sys_environment.cpp:47]: To return value use thread->retval[0]
Non zero error code will report about error to thr user space

Comment (viniciuslrangel) [reply orbis-kernel/src/sys/sys_environment.cpp]: Fixed

**Pull request merged**